Do not require AWS credentials for paasta spark-run

brenns10 opened the pull request:
For the `yelp_spam` service, we do not configure the AWS credential arguments of `paasta spark-run`. Our AWS credentials are not present on the batch machine, so it wouldn't be possible for Paasta to load them until the service is started. Instead, our service configures AWS credentials (and the resulting Spark configuration changes) itself.

When we use `paasta spark-run` (invoked by tron), Paasta fails to find AWS credentials and crashes, with the traceback below:

```
Traceback (most recent call last):
  File "/usr/bin/paasta", line 10, in <module>
    sys.exit(main())
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/paasta_tools/cli/cli.py", line 124, in main
    return_code = args.command(args)
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/paasta_tools/cli/cmds/spark_run.py", line 959, in paasta_spark_run
    system_paasta_config=system_paasta_config,
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/paasta_tools/cli/cmds/spark_run.py", line 716, in configure_and_run_docker_container
    access_key, secret_key = get_aws_credentials(args)
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/paasta_tools/cli/cmds/spark_run.py", line 404, in get_aws_credentials
    creds = Session(profile=args.aws_profile).get_credentials()
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/session.py", line 474, in get_credentials
    'credential_provider').load_credentials()
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/session.py", line 922, in get_component
    self._components[name] = factory()
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/session.py", line 179, in <lambda>
    lambda:  botocore.credentials.create_credential_resolver(self))
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/credentials.py", line 55, in create_credential_resolver
    metadata_timeout = session.get_config_variable('metadata_service_timeout')
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/session.py", line 279, in get_config_variable
    elif self._found_in_config_file(methods, var_config):
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/session.py", line 306, in _found_in_config_file
    return var_config[0] in self.get_scoped_config()
  File "/opt/venvs/paasta-tools/lib/python3.6/site-packages/botocore/session.py", line 383, in get_scoped_config
    raise ProfileNotFound(profile=profile_name)
botocore.exceptions.ProfileNotFound: The config profile (default) could not be found
```

This branch adds a `--no-aws-credentials` argument to `paasta spark-run`, and adjusts the internal workings of the command to support missing credentials. I've updated some relevant unit tests to reflect this, but would love feedback on how to end-to-end test this.

stug commented on paasta_tools/cli/cmds/spark_run.py:
```diff
@@ -257,6 +257,14 @@ def add_subparser(subparsers):
 
+    aws_group.add_argument(
+        "--no-aws-credentials",
+        help="Do not load any AWS credentials; allow the service to use its "
```
ultra-mega-nits: it's really the spark job that is doing this, right?  the service is sort of a more broad term.  Also you have the word "its" there twice

brenns10 commented:
Ok, adjusted the docstring, ran pre-commit, and did some tests, here are the cases I covered manually:
* By moving my personal AWS configuration files to backup locations, and specifying a service name which has no AWS credentials configured, I was able to replicate the traceback above from production.
* I then used the new `--no-aws-credentials` option and verified that the error no longer occurred, and the batch could start up.
* Then I replaced my AWS credentials, ran it without the new option, and saw that credentials correctly appeared in the environment
* With creds replaced, I used the new option, and saw that the creds were not in the environment

All as expected!

stug commented:
Cool, thanks!
